src/models/shelf/index.test.ts

```diff
@@ -10,7 +10,8 @@ describe('models/shelf', () => {
             x: {field: 'a', type: 'quantitative'}
           },
           anyEncodings: [],
-          config: {numberFormat: 'd'}
+          config: {numberFormat: 'd'},
+          filters: []
```

Comment:
Can you move `filters` and `transform` to be above `mark` and `encoding` for all of these?  

(Since `filters` and `transform` are applied before we encode data.)